tests/protocol/pool/Pool.Supply.Rwa.t.sol

```diff
@@ -0,0 +1,344 @@
+
+    (aUSDX, , ) = contracts.protocolDataProvider.getReserveTokensAddresses(tokenList.usdx);
+    (aBuidl, , ) = contracts.protocolDataProvider.getReserveTokensAddresses(tokenList.buidl);
+
+    vm.startPrank(poolAdmin);
+    // authorize & mint BUIDL to alice
+    buidl.authorize(alice);
+    buidl.mint(alice, 100000e6);
+    // grant Transfer Role to the aToken Transfer Admin
+    AccessControl(aclManagerAddress).grantRole(
+      keccak256('ATOKEN_TRANSFER_ROLE'),
+      aTokenTransferAdmin
+    );
+    // authorize aBUIDL contract to hold BUIDL
+    buidl.authorize(aBuidl);
+    vm.stopPrank();
+
+    vm.prank(alice);
+    buidl.approve(report.poolProxy, UINT256_MAX);
+  }
+
+  function test_first_supply() public {
+    uint256 supplyAmount = 1e6;
+    test_fuzz_first_supply(supplyAmount);
+  }
+
+  function test_fuzz_first_supply(uint256 supplyAmount) public {
+    uint256 underlyingBalanceBefore = IERC20(tokenList.buidl).balanceOf(alice);
+    supplyAmount = bound(supplyAmount, 1, underlyingBalanceBefore);
+
+    vm.expectEmit(report.poolProxy);
+    emit ReserveUsedAsCollateralEnabled(tokenList.buidl, alice);
+    vm.expectEmit(report.poolProxy);
+    emit Supply(tokenList.buidl, alice, alice, supplyAmount, 0);
+
+    vm.prank(alice);
+    contracts.poolProxy.supply(tokenList.buidl, supplyAmount, alice, 0);
+
+    assertEq(IERC20(tokenList.buidl).balanceOf(alice), underlyingBalanceBefore - supplyAmount);
+    assertEq(IAToken(aBuidl).scaledBalanceOf(alice), supplyAmount);
+  }
+
+  // supply fails if onBehalfOf does not match caller
+  function test_reverts_supply_onBehalfOfMustMatchCaller() public {
+    uint256 supplyAmount = 1e6;
+    test_fuzz_reverts_supply_onBehalfOfMustMatchCaller(supplyAmount, bob);
+  }
+
+  // fuzz - supply fails if onBehalfOf does not match caller
+  function test_fuzz_reverts_supply_onBehalfOfMustMatchCaller(
+    uint256 supplyAmount,
+    address onBehalfOf
+  ) public {
+    supplyAmount = bound(supplyAmount, 1, IERC20(tokenList.buidl).balanceOf(alice));
+    vm.assume(onBehalfOf != alice && onBehalfOf != aBuidl);
+
+    vm.expectRevert(bytes(Errors.ON_BEHALF_OF_MUST_MATCH_CALLER));
+
+    vm.prank(alice);
+    contracts.poolProxy.supply(tokenList.buidl, supplyAmount, onBehalfOf, 0);
+  }
+
+  function test_supply_after_collateral_enabled() public {
+    test_first_supply();
+    uint256 supplyAmount = 2e6;
+    uint256 underlyingBalanceBefore = IERC20(tokenList.buidl).balanceOf(alice);
+    uint256 scaledBalanceTokenBase = IAToken(aBuidl).scaledBalanceOf(alice);
+
+    vm.expectEmit(report.poolProxy);
+    emit Supply(tokenList.buidl, alice, alice, supplyAmount, 0);
+
+    vm.prank(alice);
+    contracts.poolProxy.supply(tokenList.buidl, supplyAmount, alice, 0);
+
+    assertEq(IERC20(tokenList.buidl).balanceOf(alice), underlyingBalanceBefore - supplyAmount);
+    assertEq(IAToken(aBuidl).scaledBalanceOf(alice), scaledBalanceTokenBase + supplyAmount);
+  }
+
+  function test_supplyWithPermit(
```

Comment:
shall we also mention in the test name that this is a self-permit (user calls for himself with permit)?